tests/test_ipu_configuration.py

```diff
@@ -37,7 +38,45 @@ def create_ipu_config() -> IPUConfig:
     return IPUConfig.from_dict(initial_dict)
 
 
+def create_mode_ipu_config(test_attributes: Dict[str, Any], mode):
+    helper_config = IPUConfig()
+    helper_config.mode = mode
+    ipu_config = IPUConfig(
+        **{helper_config._get_managed_attr_mode_name(attr): value for attr, value in test_attributes.items()}
+    )
+    ipu_config.mode = mode
+    return ipu_config
+
+
 class IPUConfigTester(unittest.TestCase):
+    def test_attribute_default_values(self):
+        # ipus_per_replica should equal to len(layers_per_ipu) if not provided
+        ipu_config = IPUConfig(layers_per_ipu=[1, 2, 3])
+        self.assertEqual(ipu_config.ipus_per_replica, 3)
+
+        # inference_matmul_proportion not specified but matmul_proportion is
+        ipu_config = IPUConfig(
+            layers_per_ipu=[1, 2, 3, 4],
+            matmul_proportion=[0.1, 0.2, 0.3, 0.4],
+        )
+        self.assertEqual(ipu_config.inference_matmul_proportion, [0.1, 0.2, 0.3, 0.4])
```

Comment:
This misses what was originally being tested. It is supposed to test that `matmul_proportion` _isn't_ copied if it's not compatible with `inference_layers_per_ipu`
```suggestion
        # inference_matmul_proportion not specified but matmul_proportion is
        ipu_config = IPUConfig(
            layers_per_ipu=[1, 2, 3, 4],
            inference_layers_per_ipu=[3, 7],
            matmul_proportion=[0.1, 0.2, 0.3, 0.4],
        )
        self.assertEqual(ipu_config.inference_matmul_proportion, 0.2)
```